teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -145,59 +145,170 @@ See [TEP-0074](./0074-deprecate-pipelineresources.md) for the deprecation plan f
-* [Custom scheduler](#custom-scheduler)
+* [Create a TaskGroup abstraction](#create-a-taskgroup-abstraction)
 * [Support other ways to share data (e.g. buckets)](#support-other-ways-to-share-data-eg-buckets)
-* [Focus on workspaces](#focus-on-workspaces)
+* [Task Pre and Post Steps](#task-pre-and-post-steps)
+
 
-Most of the solutions above involve allowing more than one Task to be run in the same pod, and those proposals all share
-the following pros & cons.
+### Pipeline in a Pod plus Pipelines in Pipelines
+
+In this option, the Tekton Pipelines controller constructs a pod that implements a Pipeline.
+If used with the [Pipelines in Pipelines](./0056-pipelines-in-pipelines.md) feature,
+users could choose which parts of a pipeline to run in a pod by grouping them into a sub-Pipeline.
+Any Pipelines grouped under a Pipeline executed in a pod will also execute in that pod.
 
 Pros:
-* Making it possible to execute a Pipeline in a pod will also pave the way to be able to support use cases such as
-  [local execution](https://github.com/tektoncd/pipeline/issues/235)
+* Same functionality used to run either an entire Pipeline or a sub-Pipeline in a pod.
+* Meets requirements of running multiple Tasks in one pod without external data storage.
+* Uses [an existing abstraction](https://github.com/tektoncd/community/blob/main/design-principles.md#reusability)
+  (Pipelines)
+* Pipelines already have syntax for expressing some of the features we'd likely want for this functionality, e.g.
+  `finally`
 
 Cons:
+* Using an existing abstraction (Pipelines) could be confusing if we can't support all of a Pipeline's functionality
+  when running as a pod (which is likely)
+* Need to determine a mechanism for surfacing results of individual Tasks
+
+### Pipeline executed as a TaskRun
 
-* Increased complexity around requesting the correct amount of resources when scheduling (having to look at the
-  requirements of all containers in all Tasks, esp. when they run in parallel)
-* Requires re-architecting and/or duplicating logic that currently is handled outside the pods in the controller
-  (e.g. passing results between Tasks and other variable interpolation)
+This is the approach currently taken in the
+[Pipeline to TaskRun experimental custom task](https://github.com/tektoncd/experimental/tree/main/pipeline-to-taskrun).
+In this approach, a CustomTask turns a PipelineRun into a TaskRun, and the TaskRun controller executes this TaskRun in a pod.
 
+Pros:
+* Permits reuse of existing CRDs with less re-architecting than having the PipelineRun controller run a PipelineRun in a pod.
 
-### Task composition in Pipeline Tasks
+Cons:
+* We would be [limited in the features we could support](https://github.com/tektoncd/experimental/tree/main/pipeline-to-taskrun#supported-pipeline-features)
+  to features that TaskRuns already support, or we'd have to add more Pipeline features to TaskRuns.
+* Does not surface results of Pipeline Tasks separately.
```

Comment:
I think one cons of this is approach is also that it breaks the `Task` -> `TaskRun` 1:1 relationship. 
With some extra complexity, it would be possible however to create extra `TaskRuns` and have the controller populate status and results on them.

Reply:
agreed, noted here